<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "https://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/xhtml;charset=UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=9"/>
<meta name="generator" content="Doxygen 1.8.18"/>
<meta name="viewport" content="width=device-width, initial-scale=1"/>
<title>Cancer Imaging Phenomics Toolkit (CaPTk): Getting Started</title>
<link href="tabs.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="jquery.js"></script>
<script type="text/javascript" src="dynsections.js"></script>
<link href="navtree.css" rel="stylesheet" type="text/css"/>
<script type="text/javascript" src="resize.js"></script>
<script type="text/javascript" src="navtreedata.js"></script>
<script type="text/javascript" src="navtree.js"></script>
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    extensions: ["tex2jax.js"],
    jax: ["input/TeX","output/HTML-CSS"],
});
</script>
<script type="text/javascript" async="async" src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js"></script>
<link href="doxygen.css" rel="stylesheet" type="text/css" />
</head>
<body>
<div id="top"><!-- do not remove this div, it is closed by doxygen! -->
<div id="titlearea">
<table cellspacing="0" cellpadding="0">
 <tbody>
 <tr style="height: 56px;">
  <td id="projectalign" style="padding-left: 0.5em;">
   <div id="projectname">Cancer Imaging Phenomics Toolkit (CaPTk)
   &#160;<span id="projectnumber">1.8.0.Alpha2</span>
   </div>
  </td>
 </tr>
 </tbody>
</table>
</div>
<!-- end header part -->
<!-- Generated by Doxygen 1.8.18 -->
</div><!-- top -->
<div id="side-nav" class="ui-resizable side-nav-resizable">
  <div id="nav-tree">
    <div id="nav-tree-contents">
      <div id="nav-sync" class="sync"></div>
    </div>
  </div>
  <div id="splitbar" style="-moz-user-select:none;" 
       class="ui-resizable-handle">
  </div>
</div>
<script type="text/javascript">
/* @license magnet:?xt=urn:btih:cf05388f2679ee054f2beb29a391d25f4e673ac3&amp;dn=gpl-2.0.txt GPL-v2 */
$(document).ready(function(){initNavTree('Getting_Started.html',''); initResizable(); });
/* @license-end */
</script>
<div id="doc-content">
<div class="PageDoc"><div class="header">
  <div class="headertitle">
<div class="title">Getting Started </div>  </div>
</div><!--header-->
<div class="contents">
<div class="toc"><h3>Table of Contents</h3>
<ul><li class="level1"><a href="#gs_FAQ">Frequently Asked Questions</a></li>
<li class="level1"><a href="#gs_supportedImages">Supported Images</a></li>
<li class="level1"><a href="#gs_imageLoading">Image &amp; Mask Loading</a></li>
<li class="level1"><a href="#gs_imageVisualization">Image Visualization</a></li>
<li class="level1"><a href="#gs_tabDocking">Tab Docking</a></li>
<li class="level1"><a href="#gs_seedpoints">Coordinate definition (Seed-point initialization)</a><ul><li class="level2"><a href="#gs_seedpoints_tumor">Tumor Points</a></li>
<li class="level2"><a href="#gs_seedpoints_tissue">Tissue Points</a></li>
</ul>
</li>
<li class="level1"><a href="#gs_drawing">Label Annotation/Drawing Panel</a></li>
<li class="level1"><a href="#gs_keyboard">Keyboard Shortcuts</a></li>
<li class="level1"><a href="#gs_utilities">Utilities (Command-line only)</a></li>
<li class="level1"><a href="#gs_preprocessing">Pre-processing</a></li>
<li class="level1"><a href="#gs_segmentation">Segmentation</a></li>
<li class="level1"><a href="#gs_features">Feature Extraction</a></li>
<li class="level1"><a href="#gs_specializedApps">Specialized Applications</a></li>
<li class="level1"><a href="#gs_commandLineUsage">Command Line Usage</a></li>
</ul>
</div>
<div class="textblock"><p>For Download/Installation Instructions, please see the following:</p>
<ul>
<li><a href="Download.html">Download Instructions</a></li>
<li><a href="Installation.html">Installation Instructions</a></li>
</ul>
<p>For Frequently Asked Questions, please see the dedicated <a href="FAQ.html">FAQ</a> section.</p>
<p>CaPTk has been designed as a modular platform, currently incorporating components for:</p><ol type="1">
<li>User interaction (e.g., coordinate definition, region annotation, and spherical approximation of abnormalities)<ul>
<li><a class="el" href="Getting_Started.html#gs_imageLoading">Image &amp; Mask Loading</a></li>
<li><a class="el" href="Getting_Started.html#gs_imageVisualization">Image Visualization</a></li>
<li><a class="el" href="Getting_Started.html#gs_tabDocking">Tab Docking</a></li>
<li><a class="el" href="Getting_Started.html#gs_seedpoints">Coordinate definition (Seed-point initialization)</a></li>
<li><a class="el" href="Getting_Started.html#gs_utilities">Utilities (Command-line only)</a></li>
<li><a class="el" href="Getting_Started.html#gs_drawing">Label Annotation/Drawing Panel</a></li>
<li><a class="el" href="Getting_Started.html#gs_keyboard">Keyboard Shortcuts</a></li>
</ul>
</li>
<li><a class="el" href="Getting_Started.html#gs_preprocessing">Pre-processing</a> (e.g., smoothing, bias correction, co-registration, normalization)</li>
<li><a class="el" href="Getting_Started.html#gs_segmentation">Segmentation</a></li>
<li><a class="el" href="Getting_Started.html#gs_features">Feature Extraction</a></li>
<li><a class="el" href="Getting_Started.html#gs_specializedApps">Specialized Applications</a></li>
</ol>
<p>The components of interaction and pre-processing are available for all integrated applications giving a researcher (whether clinical or computational) a single point of entry and exit for all their tasks.</p>
<p>The following sections give a high-level overview of CaPTk, with the intention of familiarizing a new user with the platform functionalities and interface.</p>
<hr  />
<h1><a class="anchor" id="gs_FAQ"></a>
Frequently Asked Questions</h1>
<p>Please see the dedicated <a href="FAQ.html">FAQ</a> section.</p>
<hr  />
<h1><a class="anchor" id="gs_supportedImages"></a>
Supported Images</h1>
<p>Currently, CaPTk supports visualization of <b>MR, CT, PET, X-Ray and Full-field Digital Mammography (FFDM) </b> images in NIfTI (i.e., <b>.nii</b>, <b>.nii.gz</b>) format. DICOM (i.e., <b>.dcm</b>) support is limited to a few protocols for MR, CT and MG modalities. In NIfTI format, for MRI, the following modalities are currently supported (the image modality box equivalent is shown in parenthesis):</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Acronym </th><th class="markdownTableHeadCenter">Image Type  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">T1 </td><td class="markdownTableBodyCenter">Native T1-weighted MR Image  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">T1Gd </td><td class="markdownTableBodyCenter">Post-contrast T1-weighted MR Image (also known as T1c, T1-CE)  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">T2 </td><td class="markdownTableBodyCenter">Native T2-weighted MR Image  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">FLAIR </td><td class="markdownTableBodyCenter">T2-weighted Fluid Attenuated Inversion Recovery MR Image  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DSC-ap-rCBV </td><td class="markdownTableBodyCenter">automatically-extracted proxy to relative Cerebral Blood Volume  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">PERFUSION </td><td class="markdownTableBodyCenter">Dynamic Susceptibility Contrast-enhanced MR Image (also known as DSC)  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DTI </td><td class="markdownTableBodyCenter">Diffusion Tensor Image  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">DWI </td><td class="markdownTableBodyCenter">Diffusion Weighted Image  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">REC </td><td class="markdownTableBodyCenter">Recurrence Maps  </td></tr>
</table>
<p><b>*</b> These modalities are currently visualized only within the "Confetti" application.</p>
<p>In addition, CaPTk offers the ability to extract and visualize commonly used measurements from DTI [1] and DSC-MRI, while accounting for leakage correction [2]. The exact measurements supported are:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Acronym </th><th class="markdownTableHeadCenter">Image Type  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DTI-FA </td><td class="markdownTableBodyCenter">Fractional Anisotropy  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">DTI-AX </td><td class="markdownTableBodyCenter">Axial Diffusivity  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DTI-RAD </td><td class="markdownTableBodyCenter">Radial Diffusivity  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">DTI-TR </td><td class="markdownTableBodyCenter">Trace  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DSC-ap-rCBV </td><td class="markdownTableBodyCenter">automatically-extracted proxy to relative Cerebral Blood Volume  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">DSC-PH </td><td class="markdownTableBodyCenter">Peak Height  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">DSC-PSR </td><td class="markdownTableBodyCenter">Percentage Signal Recovery  </td></tr>
</table>
<hr  />
<p>References:</p>
<ol type="1">
<li>J.M.Soares, P.Marques, V.Alves, N.Sousa. "A hitchhiker's guide to diffusion tensor imaging", Front Neurosci. 7:31, 2013, DOI: 10.33 fnins.2013.00031</li>
<li>E.S.Paulson, K.M.Schmainda, "Comparison of dynamic susceptibility-weighted contrast-enhanced MR methods: recommendations for measuring relative cerebral blood volume in brain tumors", Radiology. 249(2):601-613, 2008, DOI: 10.1148/radiol.2492071659</li>
</ol>
<hr  />
<h1><a class="anchor" id="gs_imageLoading"></a>
Image &amp; Mask Loading</h1>
<p>The <b>File -&gt; Load -&gt; Images</b> menu is used to load all image types. Images can also be loaded by dragging and dropping onto the application. In case of DICOM images, loading/dragging a single image loads the entire dicom series. <br  />
 Mask must be loaded via <b>File -&gt; Load -&gt; ROI</b> menu.</p>
<div class="image">
<img src="imageLoading_resize.png" alt=""/>
<div class="caption">
Loading images &amp; mask for visualization/processing</div></div>
 <hr  />
<p> For every loaded NIfTI image, the modality is automatically assigned using information from the filename. To <b> assign modalities </b> for DICOM images, or in case of a discrepancy in the modality of a NIfTI image, the user can use the drop-down modality switcher (see below) to revise the modality accordingly. This is typically needed in the <a href="ht_SpecialApps.html">Specialized Applications</a>.</p>
<div class="image">
<img src="modalities_resize.png" alt=""/>
<div class="caption">
Modality switching for individual images</div></div>
 <hr  />
<h1><a class="anchor" id="gs_imageVisualization"></a>
Image Visualization</h1>
<p>Sliders on each visualization panel (highlighted in green in the figure below) control the movement across respective axes. Note that in DSC-MRI, the single horizontal slider (highlighted in yellow) moves across the time of the acquired dynamic scan. <br  />
 Various adjustments are available to the user, including:</p><ol type="1">
<li>Window/Level: right mouse button press + horizontal/vertical mouse move on any visualization panel, respectively.</li>
<li>Manual Window/Level and setting of different visualization presets: bottom-right panel.</li>
<li>Image zoom: Ctrl + mouse wheel.</li>
<li>Image pan: Ctrl + left mouse button press + mouse move.</li>
</ol>
<p>The bottom-left panel of CaPTk (highlighted in red) shows basic information about the image and the position of the cross-hair. In the visualization panes, the "Z" axis is the center; and "Y" and "X" are to its left and right, respectively. "Z" represents the Axial view for RAI-to-LPS images.</p>
<div class="image">
<img src="sliders_resize.png" alt=""/>
<div class="caption">
Visualization Sliders and Image Information</div></div>
 <p>The "Overlay" functionality enables the visualization between two loaded images:</p>
<ol type="1">
<li>Double-click one of the two images you want to visualize and then click on its corresponding "overlay" radio button, on its right-most column.</li>
<li>Click on the "overlay" radio button of the second image that you want to visualize.</li>
<li>Both "Overlay" and "Underlay" images are now visualized with 50% opacity.</li>
<li>Check the tick box "Change Opacity".</li>
<li>Moving the slider changes the opacity between the "Overlay" and the "Underlay" images.</li>
</ol>
<p>The "Comparison Mode" puts the loaded images (3 maximum) side-by-side for easy visualization along the same axis.</p>
<div class="image">
<img src="comparisonMode.png" alt=""/>
<div class="caption">
Comparison Mode visualization</div></div>
 <hr  />
<h1><a class="anchor" id="gs_tabDocking"></a>
Tab Docking</h1>
<p>Double clicking on the tab bar will dock/undock the entire section, resulting in larger visualization panels in your monitor. This behavior is replicated by single click of the dock/undock button at the end of the section as well.</p>
<div class="image">
<img src="dock_comparison_resize.png" alt=""/>
<div class="caption">
Left is while dock is active and right is undocked behavior</div></div>
 <hr  />
<h1><a class="anchor" id="gs_seedpoints"></a>
Coordinate definition (Seed-point initialization)</h1>
<p>The <b>Seed Points</b> tab includes two general types of initialization (i.e., tumor and tissue points), controlled by the respective radio buttons.</p>
<div class="image">
<img src="seeds_resize.png" alt=""/>
<div class="caption">
Seed point initialization</div></div>
 <p>Loading and saving is done via text files in a format consistent with respective applications. Seedpoint files are needed for the following applications (and all derivatives): GLISTR (<a href="https://www.cbica.upenn.edu/sbia/software/glistr/">https://www.cbica.upenn.edu/sbia/software/glistr/</a>) [1-3], GLISTRboost (BraTS 2015 top-ranked algorithm - <a href="https://www.med.upenn.edu/sbia/glistrboost.html">https://www.med.upenn.edu/sbia/glistrboost.html</a>) [4-6], and PORTR (<a href="https://www.cbica.upenn.edu/sbia/software/portr/">https://www.cbica.upenn.edu/sbia/software/portr/</a>) [7].</p>
<p>The controls to add/remove points are the shown below.</p>
<hr  />
 <h2><a class="anchor" id="gs_seedpoints_tumor"></a>
Tumor Points</h2>
<p>These are seed-points used to approximate a tumor by a parametric spherical model, using a seed-point for its center and another for defining its radius. These are helpful for applications like tumor growth model estimation as currently incorporated in GLISTR and GLISTRboost. The controls are as follows:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Key Stroke </th><th class="markdownTableHeadCenter">Function/Action  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Shift + Space </td><td class="markdownTableBodyCenter">[SEED POINT] Set initial tumor center  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Control + Space </td><td class="markdownTableBodyCenter">[SEED POINT] Update tumor radius; on macOS, this is "Fn + Command + Space"  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Space </td><td class="markdownTableBodyCenter">[SEED POINT] Update tumor center  </td></tr>
</table>
<hr  />
 <h2><a class="anchor" id="gs_seedpoints_tissue"></a>
Tissue Points</h2>
<p>These are seed-points with coordinate information. They can be used for a multitude of applications where manual initialization(s) are required for a semi-automated algorithm, e.g, segmentation. At the moment these points are being assigned various brain tissue labels, as follows:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Tissue Acronym </th><th class="markdownTableHeadCenter">Full Form  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">CSF </td><td class="markdownTableBodyCenter">Cerebrospinal Fluid  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">VT </td><td class="markdownTableBodyCenter">Ventricular Cerebrospinal Fluid  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">GM </td><td class="markdownTableBodyCenter">Gray Matter  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">WM </td><td class="markdownTableBodyCenter">White Matter  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">VS </td><td class="markdownTableBodyCenter">Vessels  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">ED </td><td class="markdownTableBodyCenter">Edema  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">NCR </td><td class="markdownTableBodyCenter">Necrosis  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">ET </td><td class="markdownTableBodyCenter">Enhancing Tumor  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">NE </td><td class="markdownTableBodyCenter">Non-Enhancing Tumor  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">CB </td><td class="markdownTableBodyCenter">Cerebellum  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">CAE </td><td class="markdownTableBodyCenter">Enhancing Cavity  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">CAN </td><td class="markdownTableBodyCenter">Non-Enhancing Cavity  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">RTN </td><td class="markdownTableBodyCenter">Tumor Recurrence  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">RTE </td><td class="markdownTableBodyCenter">Enhanced Tumor Recurrence  </td></tr>
</table>
<hr  />
<p> Application-specific tissue types are automatically enabled when the corresponding application is selected. For example, when trying to initialize tissue points for GLISTR/GLISTRboost, only CSF, GM, WM, VS, ED, NCR, TU, NE and CB buttons will be enabled and the rest will be disabled. If there are some required tissue types missing for an application, CaPTk will display a warning and not let the user save an incomplete set of tissue points.</p>
<p><br  />
</p>
<hr  />
<p>References:</p>
<ol type="1">
<li>A.Gooya, K.M.Pohl, M.Billelo, G.Biros, C. Davatzikos, "Joint segmentation and deformable registration of brain scans guided by a tumog rowth model", Med Image Comput Comput Assist Interv. 14(Pt 2):532-40, 2011, DOI:10.1007/978-3-642-23629-7_65</li>
<li>A.Gooya, K.M.Pohl, M.Bilello, L.Cirillo, G.Biros, E.R.Melhem, C.Davatzikos, "GLISTR: glioma image segmentation and registration", IEET rans Med Imaging. 31(10):1941-54, 2012, DOI:10.1109/TMI.2012.2210558</li>
<li>D.Kwon, R.T.Shinohara, H.Akbari, C.Davatzikos, "Combining Generative Models for Multifocal Glioma Segmentation and Registration", MeI mage Comput Comput Assist Interv. 17(Pt 1):763-70, 2014, DOI:10.1007/978-3-319-10404-1_95</li>
<li>S.Bakas, K.Zeng, A.Sotiras, S.Rathore, H.Akbari, B.Gaonkar, M.Rozycki, S.Pati, C.Davatzikos, "Segmentation of gliomas in multimodam agnetic resonance imaging volumes based on a hybrid generative-discriminative framework", In Proc. Multimodal Brain Tumor Segmentation (BraTS) Challenge. 4:5-12, 2015.</li>
<li>S.Bakas, K.Zeng, A.Sotiras, S.Rathore, H.Akbari, B.Gaonkar, M.Rozycki, S.Pati, C.Davatzikos, "GLISTRboost: combining multimodal MRs egmentation, registration, and biophysical tumor growth modeling with gradient boosting machines for glioma segmentation", Brainlesion (2015). 9556:144-155, 2016, DOI:10.1007/978-3-319-30858-6_1</li>
<li>S.Bakas, H.Akbari, A.Sotiras, M.Bilello, M.Rozycki, J.Kirby, J.Freymann, K.Farahani, C.Davatzikos, "Advancing The Cancer Genome Atlag lioma MRI collections with expert segmentation labels and radiomic features", Nature Scientific Data 4:170117, 2017, DOI:10.103s data.2017.117</li>
<li>D.Kwon, M.Niethammer, H.Akbari, M.Bilello, C.Davatzikos, K.M.Pohl, "PORTR: Pre-Operative and Post-Recurrence Brain Tumor Registration", IEEE Trans Med Imaging. 33(3):651-667, 2014, DOI:10.1109/TMI.2013.2293478</li>
</ol>
<hr  />
<h1><a class="anchor" id="gs_drawing"></a>
Label Annotation/Drawing Panel</h1>
<p>This panel provides the ability to annotate Regions of Interest (ROIs) and save them as masks.</p>
<div class="image">
<img src="drawing_resize.png" alt=""/>
<div class="caption">
Drawing functions</div></div>
 <p>While drawing, you can switch across the image volumes loaded in CaPTk, by pressing the numbers 1-9 on your keyboard.</p>
<p>Toggle between non-zero (taken from the opacity box) and zero opacity of the entire annotated mask by using the keyboard key "m".</p>
<p>Visualize a single label using "Control + &lt;label number&gt;" and add labels to the visualization using "Alt + &lt;label number&gt;". To reset, press "Control/Alt + 0".</p>
<p>There are the following controls underneath the <b>Drawing</b> tab:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Button/Box </th><th class="markdownTableHeadCenter">Description  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">View Mode </td><td class="markdownTableBodyCenter">Switch from drawing mode (which is enabled when either Near or Far ROI drawing is selected) to normal viewing mode  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Line </td><td class="markdownTableBodyCenter">Draw a line of selected label color and size  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Rectangle </td><td class="markdownTableBodyCenter">Draw a rectangle of selected label color and size  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Circle </td><td class="markdownTableBodyCenter">Draw a circle of selected label color and size  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Sphere </td><td class="markdownTableBodyCenter">Draw a sphere of selected label color and pre-determined size from the size combo-box  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Free-Hand </td><td class="markdownTableBodyCenter">Draw selected label using brush of specified size using mouse left button drag  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Fill </td><td class="markdownTableBodyCenter">Fill a closed boundary with selected label - uses ITK's connected component filter  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Eraser </td><td class="markdownTableBodyCenter">Erases all labels using brush of specified size  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Label Selector </td><td class="markdownTableBodyCenter">Label selector between 1-9  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Marker Size </td><td class="markdownTableBodyCenter">A square marker of specific voxel size  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Marker Opacity </td><td class="markdownTableBodyCenter">Sets the opacity of the marker between 0.0 (transparent) and 1.0 (opaque)  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Clear Selected Label </td><td class="markdownTableBodyCenter">Remove selected label completely from the mask  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Clear All Labels </td><td class="markdownTableBodyCenter">Remove all labels completely from the mask  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Undo </td><td class="markdownTableBodyCenter">By providing the same number of old and new values in the format AxBxC, the values of the annotated region can be quickly changed  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Change Values </td><td class="markdownTableBodyCenter">By providing the same number of old and new values in the format <b>AxBxC</b>, the values of the annotated region can be quickly changed  </td></tr>
</table>
<hr  />
<h1><a class="anchor" id="gs_keyboard"></a>
Keyboard Shortcuts</h1>
<p>These are the keyboard shortcuts available on CaPTk:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Key Stroke </th><th class="markdownTableHeadCenter">Function/Action  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">1-9 </td><td class="markdownTableBodyCenter">[IMAGES] Show selected image from table  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">"m" </td><td class="markdownTableBodyCenter">[DRAWING] Show/Hide all regions; picks up last used opacity  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Control + 1-9 </td><td class="markdownTableBodyCenter">[DRAWING] Only show selected label  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Alt + 1-9 </td><td class="markdownTableBodyCenter">[DRAWING] Add selected label to visualization  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Control/Alt + 0 </td><td class="markdownTableBodyCenter">[DRAWING] Reset label visualization; i.e., show all  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Shift + Space </td><td class="markdownTableBodyCenter">[SEED POINT] Set initial tumor center  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Control + Space </td><td class="markdownTableBodyCenter">[SEED POINT] Update tumor radius; on macOS, this is "Fn + Command + Space"  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Space </td><td class="markdownTableBodyCenter">[SEED POINT] Update tumor center  </td></tr>
</table>
<hr  />
<h1><a class="anchor" id="gs_utilities"></a>
Utilities (Command-line only)</h1>
<p>To make pipeline construction using CaPTk easier, a bunch of utilities have been provided:</p>
<ol type="1">
<li>Resizing</li>
<li>Resampling</li>
<li>Image information</li>
<li>Unique values in image</li>
<li>Get smallest bounding box in mask (optional isotropic bounding box)</li>
<li>Test 2 images</li>
<li>Create mask from threshold</li>
<li>DICOM to NIfTI conversion</li>
<li>NIfTI to DICOM &amp; DICOM-Seg conversion</li>
<li>Re-orient image</li>
<li>Cast image to another pixel type</li>
<li>Thresholding:<ul>
<li>Below</li>
<li>Above</li>
<li>Above &amp; Below</li>
<li>Otsu</li>
<li>Binary</li>
</ul>
</li>
<li>Convert file formats</li>
<li>Extract Image Series from Joined stack and vice-versa (4D &lt;-&gt; 3D)</li>
<li>Transform coordinates from world to image and vice-versa</li>
<li>Label similarity Metrics between 2 label images: pure mathematical formulations are given as output</li>
<li>BraTS similarity Metrics between 2 label images: special considerations for metrics are done in this mode because output needs to be BraTS-compliant.</li>
<li>Collect information from all images in a directory and put it in CSV</li>
</ol>
<p>For a full list of this functionality and more details, please see the corresponding <a href="ht_utilities.html">How-To page</a>.</p>
<hr  />
<h1><a class="anchor" id="gs_preprocessing"></a>
Pre-processing</h1>
<p>Image pre-processing is essential to quantitative image analysis. CaPTk pre-processing tools available under the "Preprocessing" menu are fully-parameterizable and comprise:</p>
<ul>
<li><b>Denoising.</b> Intensity noise reduction in regions of uniform intensity profile is offered through a low-level image processing method, namely Smallest Univalue Segment Assimilating Nucleus (SUSAN) [1]. This is a custom implementation and does <b>NOT</b> call out to the original implementation distributed by FSL.</li>
<li><b>Co-registration.</b> Registration of various images to the same anatomical template, for examining anatomically aligned imaging signals in tandem and at the voxel level, is done using the Greedy Registration algorithm [5].</li>
<li><b>Bias correction.</b> Correction for magnetic field inhomogeneity is provided using a non-parametric non-uniform intensity normalization [2].</li>
<li><b>Intensity normalization.</b> Conversion of signals across modalities to comparable quantities using histogram matching [4].</li>
<li><b>Z-Scoring normalization.</b> Images are normalized using a z-scoring mechanism with option to do the normalization within the region of interest or across the entire image. In addition, there is an option to remove outliers &amp; noise from the image by removing a certain percentage of the top and bottom intensity ranges [6].</li>
<li><b>Histogram Matching</b></li>
<li><b>Skull Stripping (Deep Learning based)</b></li>
<li><b>Mammogram Pre-processing</b></li>
<li><b>BraTS Pre-processing Pipeline</b></li>
</ul>
<p><b>NOTE:</b> An extended set of algorithms are available via the command line utility <b>Preprocessing</b>. For full details, run the command:</p>
<pre class="fragment">Preprocessing.exe -u
</pre><div class="image">
<img src="pre-processing_resize.png" alt=""/>
<div class="caption">
Available pre-processing algorithms</div></div>
 <p><br  />
</p>
<hr  />
<p>References:</p>
<ol type="1">
<li>S.M.Smith, J.M.Brady, "SUSAN - a new approach to low level image processing", Int. J. Comput. Vis. 23(1):45-78, 1997, DOI:10.102A :1007963824710</li>
<li>N.J.Tustison, B.B.Avants, P.A.Cook, Y.Zheng, A.Egan, P.A.Yushkevich, J.C.Gee, "N4ITK: Improved N3 Bias Correction", IEEE Trans MeI maging. 29(6):1310-20, 2010, DOI: 10.1109/TMI.2010.2046908</li>
<li>S.P.Thakur, J.Doshi, S.Pati, S.M.Ha, C.Sako, S.Talbar, U.Kulkarni, C.Davatzikos, G.Erus, S.Bakas, "Brain Extraction on MRI Scans in Presence of Diffuse Glioma:
Multi-institutional Performance Evaluation of Deep Learning Methods and Robust Modality-Agnostic Training", NeuroImage 2020, DOI:10.1016/j.neuroimage.2020.117081</li>
<li>L.G.Nyul, J.K.Udupa, X.Zhang, "New Variants of a Method of MRI Scale Standardization", IEEE Trans Med Imaging. 19(2):143-50, 2000, DOI:10.1109/42.836373</li>
<li>P.A.Yushkevich, J.Pluta, H.Wang, L.E.Wisse, S.Das, D.Wolk, "Fast Automatic Segmentation of Hippocampal Subfields and Medical Temporal Lobe Subregions in 3 Tesla and 7 Tesla MRI", Alzheimer's &amp; Dementia: The Journal of Alzheimer's Association, 12(7), P126-127, DOI:10.1016/j.jalz.2016.06.205</li>
<li>T.Rohlfing, N.M.Zahr, E.V.Sullivan, A.Pfefferbaum, "The SRI24 multichannel atlas of normal adult human brain structure", Human Brain Mapping, 31(5):798-819, 2010, DOI:10.1002/hbm.20906</li>
</ol>
<hr  />
<h1><a class="anchor" id="gs_segmentation"></a>
Segmentation</h1>
<p>Currently there are two ways to produce segmentation labels for various structures in the images that are loaded:</p><ol type="1">
<li>using the Geodesic Distance Transform</li>
<li>using ITK-SNAP</li>
<li>Skull Stripping (Deep Learning based)</li>
<li>Tumor Segmentation (Deep Learning based)</li>
</ol>
<div class="image">
<img src="applicationsSeg_resize.png" alt=""/>
<div class="caption">
Segmentation applications</div></div>
 <p>More information on how to run those is provided in the <a href="ht_Segmentation.html">corresponding How-To section</a>.</p>
<hr  />
<h1><a class="anchor" id="gs_features"></a>
Feature Extraction</h1>
<p>The feature extraction tab in CaPTk enables clinicians and other researchers to easily extract feature measurements, commonly used in image analysis, and conduct large-scale analyses in a repeatable manner.</p>
<div class="image">
<img src="featurepanel_resize.png" alt=""/>
<div class="caption">
Feature Panel screenshot</div></div>
 <p>Although the feature panel in CaPTk is continuously expanding, it currently comprises i) intensity-based, ii) textural (GLCM, GLRLM, GLSZM, NGTDM, LBP), and iii) volumetric/morphologic features. You can find the exact list of features incorporated in CaPTk, together with their mathematical definition in the <a href="ht_FeatureExtraction.html">corresponding How-To section</a>.</p>
<p>Specialized applications in CaPTk, such as "EGFRvIII Surrogate Index", "Survival Prediction", "Recurrence Estimator", and "SBRT-Lung" use features of this panel. However, the general idea is to keep the features generic and adaptable for different types of images by just changing the input parameters. Currently we provide some pre-selected parameters for Neuro and Torso images (i.e., Brain, Breast, Lung). Users can alter these pre-selected values through the <b>Custom</b> menu option, or create their own set of parameters via the <b>Advanced</b> menu. The <b>output</b> of the feature extraction tab is a text (<b>.csv</b>) file, with feature names and values. Note that the reported features are extracted per modality, per annotated region and per offset (offset represents the radius around the center pixel; for radius 1, the offset will be +/- 1) value.</p>
<hr  />
<h1><a class="anchor" id="gs_specializedApps"></a>
Specialized Applications</h1>
<p>Various <a href="ht_SpecialApps.html">Specialized Applications</a> are currently incorporated in CaPTk focusing on brain tumors, breast cancer, and lung nodules as shown in the figure below.</p>
<div class="image">
<img src="applications_resize.png" alt=""/>
<div class="caption">
Feature Panel screenshot</div></div>
 <p>For details, please see <a class="el" href="ht_SpecialApps.html">Specialized Applications (SAs) Usage</a>.</p>
<hr  />
<h1><a class="anchor" id="gs_commandLineUsage"></a>
Command Line Usage</h1>
<p>Almost every application offered by CaPTk can be used from the command line and utilizing the CaPTk's powerful command line APIs is relatively straight-forward. The following parameters are present in every CLI:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Laconic </th><th class="markdownTableHeadCenter">Verbose </th><th class="markdownTableHeadCenter">Function/Action  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">u </td><td class="markdownTableBodyCenter">usage </td><td class="markdownTableBodyCenter">Usage message giving all the possible parameters of an application  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">h </td><td class="markdownTableBodyCenter">help </td><td class="markdownTableBodyCenter">Help message giving details of parameters (including defaults and ranges) with some examples  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">v </td><td class="markdownTableBodyCenter">version </td><td class="markdownTableBodyCenter">The version information of the application  </td></tr>
</table>
<p>An example to show the verbose help for an application called <b>${Application}</b>:</p>
<table class="markdownTable">
<tr class="markdownTableHead">
<th class="markdownTableHeadCenter">Platform (x64) </th><th class="markdownTableHeadCenter">Command  </th></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">Windows </td><td class="markdownTableBodyCenter">${CaPTk_InstallDir}/bin/${ApplicationName}.exe -h  </td></tr>
<tr class="markdownTableRowEven">
<td class="markdownTableBodyCenter">Linux </td><td class="markdownTableBodyCenter">${CaPTk_InstallDir}/bin/${ApplicationName} -h  </td></tr>
<tr class="markdownTableRowOdd">
<td class="markdownTableBodyCenter">macOS </td><td class="markdownTableBodyCenter">~/Applications/CaPTk_${version}.app/Contents/Resources/bin/${ApplicationName} -h  </td></tr>
</table>
<p>Detailed explanation of using the command line is available in the <a class="el" href="How_To_Guides.html">How To Guides</a>.</p>
<hr  />
 </div></div><!-- contents -->
</div><!-- PageDoc -->
</div><!-- doc-content -->
<!-- start footer part -->
		<div id="nav-path" class="navpath"><!-- id is needed for treeview function! -->
			<ul>
				<class="footer">
					Generated by <a href="http://www.doxygen.org/index.html">
					<img class="footer" src="doxygen.png" alt="doxygen"/></a>.
			</ul>
		</div>
		<script src="custom.js"></script>
	</body>
</html>
